.github/workflows/create-tag.yml

```
@@ -13,38 +13,42 @@ jobs:
       contents: write
       packages: write
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           token: ${{ secrets.CI_TOKEN }}
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v4
+
+      - name: Set up Conda environment with Micromamba
+        uses: mamba-org/setup-micromamba@v2
         with:
-          python-version: "3.10"
+          cache-environment: true
+          create-args: >-
+            python=3.10
+          environment-file: conda-environment-dev.yml
+          post-cleanup: all
 
       - name: Update version
         id: update-version
         run: |
-            pip install --upgrade pip
-            pip install poetry
             poetry version minor
```

We might need something like this to use the Poetry command installed within the Conda environment:

```suggestion
            ${MAMBA_EXE} run --name mxcubecore poetry version minor
```

Or maybe it is possible to modify the `PATH` environment variable. I do not recall if I tried this.

This workflow is near impossible to try in isolation (without actually creating tags)... so maybe it is simpler to revert back to your first suggestion... :/

---

You are right, for some reason, I thought that setup-micromamba would automatically activate the environment. So I added this before every poetry command.

I also tested the latest change here on a fork, where it seems to work including the version tag creation on github and the package build. I did not test the part that publishes to PyPI, as it would require some more setup

---

I think `setup-micromamba` does not activate the environment. I recall it being a bit weird to activate the environment, but it was quite a long time ago, maybe things are better now.